### ISSUE TITLE
Llama-CPP Install Issue - Windows

### ISSUE THREAD

**Issue by ElliottDyson** [intel-analytics/ipex-llm#10820]
The newest drivers are in use, the system is a Ryzen 2700x CPU with 16GB of RAM and a 16GB A770 GPU on Windows 11.

The instructions in the docs were followed precisely.

Upon attempting to execute main.exe (either with -h or when launching a .gguf file) we get a system error stating that code execution couldn't proceed because "mkl_sycl_blas.4.dll was not found". I have tried wiping the folder and reactivating the batch file but with no change.

Some help would be much appreciated. Thank you.

**Comment by rnwang04** [comment]
Hi @ElliottDyson 
This error (`mkl_sycl_blas.4.dll was not found`) is caused by missing of oneapi.
Have you rightly install oneapi 2024.0 and source it by `call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"` (if you use offline installed oneAPI) ?

**Comment by ElliottDyson** [comment]
> Hi @ElliottDyson 
> This error (`mkl_sycl_blas.4.dll was not found`) is caused by missing of oneapi.
> Have you rightly install oneapi 2024.0 and source it by `call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"` (if you use offline installed oneAPI) ?

This is definitely odd behaviour. I have done this yes, and the call printed out the correct statements too. I know I can usually get this to work properly as I have a separate environment for running models directly using the transformers library by huggingface and that also seems to work as I'm using imports from Ipex-LLM without error.

**Comment by rnwang04** [comment]
Hi @ElliottDyson ,
Suppose you have two conda enviroments, and just named them `llm-cpp` for cpp-based and `llm` for  transformers-based code,
Is your question why you can run the program directly in the `llm` environment, but need to manually `call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"` in the `llm-cpp` environment?
If so, the problem I guess is you are not using pip installer to install OneAPI in your `llm-cpp` conda env.
Just try with `pip install dpcpp-cpp-rt==2024.0.2 mkl-dpcpp==2024.0.0 onednn==2024.0.0` in your `llm-cpp` conda env,
then there is no need to `call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"`  manully. And don't forget to execute main.exe under the `llm-cpp` env.

**Comment by ElliottDyson** [comment]
> Hi @ElliottDyson ,
> Suppose you have two conda enviroments, and just named them `llm-cpp` for cpp-based and `llm` for  transformers-based code,
> Is your question why you can run the program directly in the `llm` environment, but need to manually `call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"` in the `llm-cpp` environment?
> If so, the problem I guess is you are not using pip installer to install OneAPI in your `llm-cpp` conda env.
> Just try with `pip install dpcpp-cpp-rt==2024.0.2 mkl-dpcpp==2024.0.0 onednn==2024.0.0` in your `llm-cpp` conda env,
> then there is no need to `call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"`  manully. And don't forget to execute main.exe under the `llm-cpp` env.
> 

No, I only mentioned the LLM environment to help explain that me following the installation process, my drivers, oneAPI, and visual studio, are not the causes of the problems being encountered in the llm-cpp environment. Everything to do with llama cpp is being done in the llm-cpp environment. Sorry for the confusion.
